static/js/app.ide.js

```diff
@@ -134,27 +153,28 @@ export default class IDEApp extends App {
    */
   async setEditorFileContent(editorComponent, clearUndoStack = false) {
     const filepath = editorComponent.getPath();
+    if (!filepath) return;
 
-    if (!filepath || (filepath && !(await this.vfs.pathExists(filepath)))) {
-      return;
-    }
-
-    // Check if the editor exceeded the maximum allowed file size.
-    const size = await this.vfs.getFileSizeByPath(filepath);
-    if (size > MAX_FILE_SIZE) {
-      editorComponent.exceededFileSize();
-      return;
-    }
+    try {
+      const content = await this.vfs.readFile(filepath, { MAX_FILE_SIZE });
 
-    const content = await this.vfs.getFileContentByPath(filepath);
-    editorComponent.setContent(content);
+      editorComponent.setContent(content);
 
-    const cursorPos = editorComponent.getCursorPosition();
-    editorComponent.setContent(content);
-    editorComponent.setCursorPosition(cursorPos);
+      const cursorPos = editorComponent.getCursorPosition();
+      editorComponent.setContent(content);
+      editorComponent.setCursorPosition(cursorPos);
 
-    if (clearUndoStack) {
-      editorComponent.clearUndoStack();
+      if (clearUndoStack) {
+        editorComponent.clearUndoStack();
+      }
+    } catch (err) {
+      if (err instanceof FileTooLargeError) {
```

Comment:
Errors in een worker throwen kan je niet opvangen, daarom gebruik ik in andere workers callback functies en vang deze op in de bridge class tussen de app en de worker om te bepalen wat ik doe als er een error vanuit de worker komt.

Reply:
Er zit een mechanisme in de worker om exceptions op te vangen en als postMessage naar de vfs-client door te sturen, die er dan weer een exception van maakt :-) Dit is waarschijnlijk iets te overdreven, ik zal nog ff kijken naar je voorbeeld in andere worker. Kun je een regel aanwijzen in een andere worker waar je een foutmelding communiceert?

Reply:
> Er zit een mechanisme in de worker om exceptions op te vangen en als postMessage naar de vfs-client door te sturen, die er dan weer een exception van maakt :-) Dit is waarschijnlijk iets te overdreven

Precies, dat is wat dubbel op. Ik zou dan gewoon met callbacks werken zoals het al overal gedaan wordt.

> Kun je een regel aanwijzen in een andere worker waar je een foutmelding communiceert?

https://github.com/uvapl/terra/blob/main/static/js/workers/git.worker.js#L213

Reply:
Deze heb ik nog even laten hangen, is nu niet zo belangrijk voor de werking en dan kan ik op de achtergrond even nadenken over wat netjes is hier.